sql/sql_connect.cc

```diff
@@ -84,6 +84,20 @@ static int increment_connection_count(THD* thd, bool use_lock);
 
 static HASH hash_user_connections;
 
+int get_user_con_by_name(const char *user) {
```

Comment:
```
  /*
     Pointer to user+host key (pair separated by '\0') defining the entity
     for which resources are counted (By default it is user account thus
     priv_user/priv_host pair is used. If --old-style-user-limits option
     is enabled, resources are counted for each user+host separately).
  */

  user_len= strlen(user);
  temp_len= (strmov(strmov(temp_user, user)+1, host) - temp_user)+1;
  mysql_mutex_lock(&LOCK_user_conn);
  if (!(uc = (struct  user_conn *) my_hash_search(&hash_user_connections,
                                               (uchar*) temp_user, temp_len)))
```

userstats using:
```
typedef struct st_user_stats {
  char user[max(USERNAME_LENGTH, LIST_PROCESS_HOST_LEN) + 1];

but:
  // Update by user name
  if ((user_stats = (USER_STATS *) my_hash_search(&global_user_stats,
                                                  (uchar *) user_string,
                                                  strlen(user_string))))

Where name:
  char* user_string=         get_valid_user_string(thd->main_security_ctx.user);

static char* get_valid_user_string(char* user) {
  return user ? user : mysql_system_user;
}
```

thd->main_security_ctx.user looks to be just a username without host information and this is confusing with USER_STATS->user:   char user[max(USERNAME_LENGTH, LIST_PROCESS_HOST_LEN) + 1]; definition.

Should I return back to independent concurrent connections calculation?
or fill USER_STATS->user in the same way as user_conn hash key?